mono/mini/mini-amd64.c

```diff
@@ -392,6 +392,20 @@ add_valuetype_win64 (MonoMethodSignature *sig, ArgInfo *ainfo, MonoType *type,
 
 	klass = mono_class_from_mono_type (type);
 	size = mini_type_stack_size_full (&klass->byval_arg, NULL, sig->pinvoke);
+
+	ainfo->pass_empty_struct = FALSE;
+#ifndef __GNUC__
+	/*
```

Comment:
If we are OK to change the behavior for GCC build modules, we can move over to follow the C standard around empty structs, but then we will be incompatible with GCC/Clang build modules using the C extension. This could be a generic problem on all platforms using different compilers for runtime and modules, so the best would be if runtime could detect this per module and handle both cases depending on implementation (base on meta data associated with module). But that is probably a bigger task that would need some more use cases that this could piggy back on. Will just apply this for x64 Windows (follow the C standard way on both GCC and MSVC builds) for now to minimize potential regression.